MonoGame.Framework/Windows/WinFormsGameWindow.cs

```diff
@@ -323,9 +324,13 @@ private void OnMouseLeave(object sender, EventArgs e)
             }
         }
 
+        [DllImport("user32.dll")]
+        private static extern short VkKeyScanEx(char ch, IntPtr dwhkl);
+
         private void OnKeyPress(object sender, KeyPressEventArgs e)
         {
-            OnTextInput(sender, new TextInputEventArgs(e.KeyChar));
+            var key = (Keys) (VkKeyScanEx(e.KeyChar, InputLanguage.CurrentInputLanguage.Handle) & 0xff);
```

Comment:
Wait, this works O o, never realized that XNA Keys enum was based off of this one.

Anyway, are you sure there is no better way of doing this like abusing other events to get more accurate information like I did for UWP?

Reply:
I mentioned an alternative in my original post:

> The alternative is to store the key pressed in WinForms' KeyDown event as was done in #4931 , but that isn't foolproof either and more hacky to implement.

We just have to choose either one. Neither solution is great, but both are okay in most cases.

Reply:
Oh, I missed that, well we could have both implemented and see what the users complain about and figure out the better way that way (I mean different implementations for UWP and DX versions).

Reply:
Heh, that's actually pretty clever :D